src/main/java/com/iota/iri/zmq/ZmqTransactionStatsPublisher.java

```diff
@@ -0,0 +1,134 @@
+
+    private final Tangle tangle;
+    private final TipsViewModel tipsViewModel;
+    private final TipSelector tipsSelector;
+    private final TimeWindowedApproveeCounter approveeCounter;
+
+    private final MessageQ messageQ;
+
+    private final AtomicBoolean shuttingDown = new AtomicBoolean(false);
+    private Thread thread;
+
+    public ZmqTransactionStatsPublisher(Tangle tangle, TipsViewModel tipsViewModel, TipSelector tipsSelector,
+            MessageQ messageQ) {
+
+        this.tangle = tangle;
+        this.tipsViewModel = tipsViewModel;
+        this.tipsSelector = tipsSelector;
+        this.approveeCounter = new TimeWindowedApproveeCounter(tangle, MIN_TRANSACTION_AGE_THRESHOLD,
+                MAX_TRANSACTION_AGE_THRESHOLD);
+        this.messageQ = messageQ;
+    }
+
+    /**
+     * Starts the publisher.
+     */
+    public void init() {
+        thread = new Thread(getRunnable(), "Transaction Stats Publisher");
+        thread.start();
+    }
+
+    private Runnable getRunnable() {
+        return () -> {
+            while (!shuttingDown.get()) {
+                try {
+                    // the transaction arrival time is in seconds
+                    final long now = Instant.now().getEpochSecond();
+
+                    final long numConfirmed = getConfirmedTransactionsCount(now);
+                    final long numTransactions = getAllTransactionsCount(now);
+
+                    messageQ.publish(CONFIRMED_TRANSACTIONS_TOPIC + " %d", numConfirmed);
+                    messageQ.publish(TOTAL_TRANSACTIONS_TOPIC + " %d", numTransactions);
+                } catch (Exception e) {
+                    log.error("Error while getting transaction counts : {}", e);
+                }
+                try {
+                    Thread.sleep(PUBLISH_INTERVAL);
+                } catch (InterruptedException e) {
+                    log.error("Transaction count interrupted.");
+                }
+            }
+        };
+    }
+
+    private Hash getSuperTip() throws Exception {
+
+        // call the usual tip selection and return the first tip
+        List<Hash> tips = tipsSelector.getTransactionsToApprove(Optional.empty());
+
+        return tips.get(0);
+    }
+
+    private long getConfirmedTransactionsCount(long now) throws Exception {
+
+        return approveeCounter.getCount(now, getSuperTip(), new HashSet<>());
+    }
+
+    private long getAllTransactionsCount(long now) throws Exception {
+
+        // count all transactions in a scalable way, by counting the approvees of all the tips
+        HashSet<Hash> processedTransactions = new HashSet<>();
+        long count = 0;
+        for (Hash tip : tipsViewModel.getTips()) {
+            // count the tip, if it is the valid time window
+            if (approveeCounter.isInTimeWindow(now, TransactionViewModel.fromHash(tangle, tip))) {
+                count++;
+            }
+            count += approveeCounter.getCount(now, tip, processedTransactions);
```

Comment:
Suggestion: move this line inside `if` statement, and change `count++` to 
`count += 1 + approveeCounter.getCount(now, tip, processedTransactions);`